### ISSUE TITLE
PATH quotes and spaces wierdness

### ISSUE THREAD

**Issue by BurtHarris** [Microsoft/BashOnWindows#2155]
* Your Windows build number:  10.0.15063
* What's going wrong:  When invoking `bash`  from a windows command prompt, the translation of the windows PATH environment variable is befuddled by quotes and/or spaces.    

1.  Quoted paths in the PATH are dropped from bash's PATH :

```
C:\Users\nonpriv>path c:\Windows\System32;"c:\Program Files\nodejs"
C:\Users\nonpriv>node -v
v6.10.3
C:\Users\nonpriv>bash -c printenv | findstr PATH
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/mnt/c/Windows/System32
C:\Users\nonpriv>bash -c node -v
-v: node: command not found
```

2. Non-quoted paths containing spaces do appear inside the bash environment, but don't work properly:

```
C:\Users\nonpriv>path c:\Windows\System32;c:\Program Files\nodejs

C:\Users\nonpriv>bash -c printenv | findstr PATH
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/mnt/c/Windows/System32:/mnt/c/Program Files/nodejs

C:\Users\nonpriv>bash -c node -v
-v: node: command not found
```

**Comment by benhillis** [comment]
Does bash.exe -c "node -v" work?

**Comment by BurtHarris** [comment]
No, in both cases that generates `/bin/bash: node: command not found`

Thanks for catching that though.

**Comment by therealkenc** [comment]
...Stumbled across in a dupe search.

> C:\Users\nonpriv>path c:\Windows\System32;"c:\Program Files\nodejs"

That scenario is pretty legit. Confirmed the Windows quoted paths, which are allowed, don't get pulled into WSL's `$PATH` (circa 19551). Work around: don't put quotes in a Windows path.

> C:\Users\nonpriv>bash -c node -v

You want (in modern parlance):

```
$ wsl.exe node.exe -v
```

![image](https://user-images.githubusercontent.com/7049700/73174896-c0734d80-40bd-11ea-85e9-90e7b79e67e4.png)

Windows is happy to assume the executable has a `.exe` extension. Linux is picky about the name being exact by-design, and has nothing to do with the spaces in the `$PATH`.

Tweaking title for the first part.
